Assets/Scripts/ArenasParameters.cs

```diff
@@ -141,6 +141,7 @@ public class ArenaConfiguration
 		public bool toUpdate = false;
 		public string protoString = "";
 		public int randomSeed = 0;
+		public bool mergeNextArena = false;
```

Comment:
I think the second of your points re flexibility is what's important for this change.

An example use case for this feature is to have a maze that the agent navigates several times during one episode. For example the first time might be to learn the maze structure and then the second time we've added a block, so that the most efficient route is no longer possible and the agent has to use its learning from the previous run to reroute. For this example if we had 3 mazes we wanted to test on, we would have 6 arenas we want to merge in pairs (0-1, 2-3, 4-5).

Reply:
In this case, perhaps an elaboration could be made on the actual and most probable use case for this feature, in the docs. Maybe it was just me but i did not understand why and how it would be used :)